File: SelectionSort.js
// [3, 44, 38, 5, 47, 15, 36, 26, 27, 2, 46, 4, 19, 50, 48]

function selectionSort(array) {

    for (let i = 0; i < array.length; i++) {

        let min = i;

        for (let j = i + 1; j < array.length; j++) {

            if (array[min] > array[j]) {
                min = j;
            }

        }
        [array[i], array[min]] = [array[min], array[i]]
    }
    return array
}

let result = selectionSort([3, 44, 38, 5, 47, 15, 36, 26, 27, 2, 46, 4, 19, 50, 48])
console.log(result)

File: sorting/selectionSort/selectionSort.py
def selectionSort(array):
    for i in range(0, len(array)):
        min = i
        for j in range(i+1 , len(array)):
            if array[min] > array[j]:
                min = j
        array[i] , array[min] = array[min] , array[i]
    return array


result = selectionSort([0, 2, 34, 22, 10, 19])
print(result)
